### PR TITLE
Upgrade spotless and use runToFixMessage for 'gradlew tidy' hint.

### PR DESCRIPTION
**Pull request by dweiss**

### CODE REVIEW

**Comment by dweiss** [comment]
I removed the hack and updated spotless. Works fine for me. I just noticed that any gradle error now returns the 'something went wrong' message though. 

image

File: gradle/validation/spotless.gradle [hunk 1]
```diff
@@ -110,10 +110,8 @@ configure(project(":lucene").subprojects) { prj ->
     check.dependsOn v
     v.dependsOn ":checkJdkInternalsExportedToGradle"
   }
-}
 
-gradle.taskGraph.afterTask { Task task, TaskState state ->
-  if (task.name == 'spotlessJavaCheck' && state.failure) {
-    throw new GradleException("\n****************************\n*PLEASE RUN ./gradlew tidy!*\n****************************");
+  tasks.matching { task -> task.name == "spotlessJavaCheck" }.configureEach {
```
Comment (uschindler) [inline gradle/validation/spotless.gradle:114]: Should we not move this above where the other tasks are configured?

Comment (dweiss) [reply gradle/validation/spotless.gradle]: I moved it up.

**Comment by rmuir** [comment]
> I just noticed that any gradle error now returns the 'something went wrong' message though.

This only happens on windows. Something is off with the `%ERRORLEVEL%` stuff in the .bat file I think.

**Comment by dweiss** [comment]
Will take a look

**Comment by rmuir** [comment]
I took a look, it seems to me at a lance:
* any error from WrapperDownloader will go to `:fail` section via an explicit GOTO: `IF %ERRORLEVEL% NEQ 0 goto fail`
* any error from gradlew itself will fall through to `:fail` section simply because it is located above it?

**Comment by dweiss** [comment]
This fixes the script logic for windows: https://github.com/apache/lucene/pull/835

File: gradle/validation/spotless.gradle (hunk 1 above)
Comment (uschindler) [reply gradle/validation/spotless.gradle]: Thats also fine. I meant to move it up to the configuration logic next to "spotlessJava" task (or is this the extensions?). No worries - all fine. I am just a bit against those "tasks.matching", "tasks.withType" everywhere and would rather be explicit with Gradle DSL syntax.

Comment (dweiss) [reply gradle/validation/spotless.gradle]: Ah... I see what you mean. No, this won't work. This property is defined on a concrete task type and not on the DSL extension (from which the various spotless tasks are dynamically created).

Comment (dweiss) [reply gradle/validation/spotless.gradle]: Those tasks.matching and withType are often required - they're dynamic collections and invoke the configuration block for objects that satisfy the criteria but are created later (or elsewhere) during configuration. It's a way to break dependency cycles between script evaluation, plugins, task creation, etc. I don't think there is a nicer way to do many of the things these blocks are currently used for.

**Review by uschindler — APPROVED** [review]

**Comment by dweiss** [comment]
Thanks @uschindler

**Comment by rmuir** [comment]
Thank you for patching spotless and integrating the fix @dweiss !

**Pull request merged**